### ISSUE TITLE
Semicolons (at end of line) in KDoc aren't allowed on lines containing references, but should be

### ISSUE THREAD

**Issue by erikhuizinga** [pinterest/ktlint#362]
Ktlint doesn't accept the following:

```kotlin
/**
 * [x];
 */
```

Ktlint doesn't allow semicolons at the end of a KDoc line. However, they are perfectly fine, as they are a natural language token, not a Kotlin token.

It seems this only happens when the KDoc contains references to code, e.g..: `[x]`.
Note that just formatted code with a semicolon, e.g. `` `x`;``, is accepted by ktlint.

The suggestion is to accept EOL semicolons in KDoc lines that contain Kotlin code.

**Comment by NightlyNexus** [comment]
I couldn't reproduce this error.
Can you make for `NoSemicolonsRuleTest` or link to a minimum reproducible example?

**Comment by erikhuizinga** [comment]
You're right, ktlint runs without problems on my original example. I'm editing it right away! Save it to `file.kt` and run `ktlint file.kt`, which results in `/path/to/file.kt:2:7: Unnecessary semicolon`.
